test/parallel/test-assert-async.js

```diff
@@ -27,211 +29,192 @@ const invalidThenableFunc = () => {
-    assert.rejects(Promise.reject(err), { code: 'FOO' }, 'AAAAA'),
-    handler.bind(null, false, err)
-  ));
-}
-
-// Check `assert.doesNotReject`.
-{
-  // `assert.doesNotReject` accepts a function or a promise
-  // or a thenable as first argument.
-  /* eslint-disable no-restricted-syntax */
-  let promise = assert.doesNotReject(() => new Map(), common.mustNotCall());
-  promises.push(assert.rejects(promise, {
-    message: 'Expected instance of Promise to be returned ' +
-             'from the "promiseFn" function but got an instance of Map.',
-    code: 'ERR_INVALID_RETURN_VALUE',
-    name: 'TypeError'
-  }));
-  promises.push(assert.doesNotReject(async () => {}));
-  promises.push(assert.doesNotReject(Promise.resolve()));
-
-  // `assert.doesNotReject` should not accept thenables that
-  // use a function as `obj` and that have no `catch` handler.
-  const validFulfillingThenable = {
-    then: (fulfill, reject) => {
-      fulfill();
-    },
-    catch: () => {}
-  };
-  promises.push(assert.doesNotReject(validFulfillingThenable));
-  promises.push(assert.rejects(
-    assert.doesNotReject(invalidThenable),
-    {
-      code: 'ERR_INVALID_ARG_TYPE'
-    })
-  );
-  promises.push(assert.rejects(
-    assert.doesNotReject(invalidThenableFunc),
-    {
-      code: 'ERR_INVALID_RETURN_VALUE'
-    })
-  );
-
-  const handler1 = (err) => {
-    assert(err instanceof assert.AssertionError,
-           `${err.name} is not instance of AssertionError`);
-    assert.strictEqual(err.code, 'ERR_ASSERTION');
-    assert.strictEqual(err.message, 'Failed');
-    return true;
-  };
-  const handler2 = (err) => {
-    assert(err instanceof assert.AssertionError,
-           `${err.name} is not instance of AssertionError`);
-    assert.strictEqual(err.code, 'ERR_ASSERTION');
-    assert.strictEqual(err.message,
-                       'Got unwanted rejection.\nActual message: "Failed"');
-    assert.strictEqual(err.operator, 'doesNotReject');
-    assert.ok(err.stack);
-    assert.ok(!err.stack.includes('at Function.doesNotReject'));
-    return true;
-  };
-
-  const rejectingFn = async () => assert.fail();
-
-  promise = assert.doesNotReject(rejectingFn, common.mustCall(handler1));
-  promises.push(assert.rejects(promise, common.mustCall(handler2)));
-
-  promise = assert.doesNotReject(rejectingFn(), common.mustCall(handler1));
-  promises.push(assert.rejects(promise, common.mustCall(handler2)));
-
-  promise = assert.doesNotReject(() => assert.fail(), common.mustNotCall());
-  promises.push(assert.rejects(promise, common.mustCall(handler1)));
-
-  promises.push(assert.rejects(
-    assert.doesNotReject(123),
-    {
+  it('should not accept thenables that use a function as `obj` and that have no `catch` handler', async () => {
+    await assert.rejects(assert.rejects(invalidThenable, {}), {
+      code: 'ERR_INVALID_ARG_TYPE',
+    });
+  });
+
+  it('The validate validation function is expected to return "true"', async () => {
```

Comment:
```suggestion
  it('the validate validation function is expected to return "true"', async () => {
```
Nit: for consistency.